src/library_wasi.js

```diff
@@ -33,7 +33,8 @@ var WasiLibrary = {
       };
       // Apply the user-provided values, if any.
       for (var x in ENV) {
-        env[x] = ENV[x];
+        if (ENV[x] === undefined) delete env[x];
```

Comment:
This is slightly subtle, so a comment may be useful, perhaps
```suggestion
        // We loop on ENV, and so if ENV[x] is undefined that means the
        // object has that key, and the value is undefined (as opposed to
        // getting undefined when the key is not present). A present key
        // with undefined indicates to us to delete the key.
        if (ENV[x] === undefined) delete env[x];
```